### ISSUE TITLE
Leo Crashes On Automatic Restart

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#3141]
See PR #3879.

The `restart-leo` command does not work when Leo was started from a script.

**Original report**

This has happened to me twice in a row now.  I update my git repo and get a notice from Leo that one of the open outlines has changed.  It asks if I want to restart,  and I click "yes".  Leo restarts, but never gets beyond the splash screen.  CTRL-C brings back control to the console but does not clear the spash screen and Leo won't restart.  If instead I forcibly close the terminal, the splash screen vanishes and the next time I launch Leo from a new terminal it launches normally.

```
Leo 6.7.2-devel, devel branch, build 7d0eb2c4b0
2023-02-10 04:07:08 -0600
Python 3.10.9, PyQt version 6.4.1
Windows 10 AMD64 (build 10.0.19044) SP0
```

I have not tried to find out the cause as yet.  The Windows Event Viewer shows no relevant events.

**Comment by tbpassin** [comment]
This issue occurs running the _restart-leo_ command too (not a surprise). 
It only happens when I start Leo from my batch file that sets the PYTHONPATH to point to my Leo clone.  It does not happen when I cd to the git\leo-editor directory and then start Leo from the console with `py -m leo.core.runLeo`.

So it must be some kind of interaction between the `execve()`  that restarts Leo, and the evvironment set up by the batch file I'm running it from, since the `execve()` call invokes python.exe directly, rather than the batch file I originally started it with.

**Comment by tbpassin** [comment]
Quasi-solved.  While the relaunched Leo is hung displaying its splash screen, if I hit <ENTER> in the console then Leo continues on to complete its startup.  Apparently it gets into some state where it is waiting for the console to read a line.

I have no idea why, but at least it's nothing too serious. it only happens when I launch Leo from my batch file.

**Comment by edreamleo** [comment]
@tbpassin Good work, Thomas. Many thanks for your comments and investigations.

**Comment by tbpassin** [comment]
I'm not sure Leo is completely off the hook yet, because I now remember that when I have shut it down in the past (I mean without restart), that its launching console sometimes also needed me to press \<ENTER\> before it would display the prompt and be usable again.  That's probably the same phenomena, somehow related to my batch file.

The batch file is fairly simple.  

```
setlocal
set PYTHONPATH=....
py -m leo.core.runLeo %*
endlocal
```

It would also work if I `cd`-ed to the git\leo-editor directory instead of setting the PYTHONPATH, but I don't like for a batch file to change directories out from under me if it's not necessary.

Or maybe it's a QT thing.

There's still room here for more nosing around.

**Comment by edreamleo** [comment]
@tbpassin I am going to delegate this issue to you and remove the milestone. Please feel free to close this issue.

**Comment by LewisNeal** [comment]
Using the restart-leo command resumes the console with a triple arrow prompt
>>>
and not the usual single arrow prompt in which I started the Leo session:
leo-editor>

_Leo 6.7.4-devel, devel branch, build cb648f254a
2023-06-19 08:34:47 -0500
Python 3.11.3, PyQt version 6.5.1
Windows 10 AMD64 (build 10.0.19044) SP0_

**Comment by LewisNeal** [comment]
For some reason the triple prompt '>>>' was deleted from my post :(

**Comment by LewisNeal** [comment]
Here is a log of console with Leo stuck and not restarting:

wrote recent file: C:/Users/lewisneal/.leo/.leoRecentFiles.txt
restarting Leo...
wrote C:/Users/lewisneal/.leo/leo.session
Python 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
'>>>'
N:\git\leo-editor>

**Comment by tbpassin** [comment]
Possibly the same issue as https://github.com/leo-editor/leo-editor/issues/3141

**Comment by LewisNeal** [comment]
On windows Leo does not restart - pressing <Enter> just toggles the prompt type as shown:
`>>>`
N:\git\leo-editor>
`>>>`
N:\git\leo-editor>

**Comment by LewisNeal** [comment]
Start Leo in the same console session and then repeat the 'restart-leo' command. Press `<Enter>` repeatedly and it shows a changed pattern in the console prompt:
`>>>`
N:\git\leo-editor>
`>>>`
`>>>`
N:\git\leo-editor>
`>>>`
`>>>`

**Comment by edreamleo** [comment]
@tbpassin Thanks for your continuing work.

**Comment by tbpassin** [comment]
If Leo is run using the Windows leo.exe executable in the Python3.xx\Scripts directory, the restart-leo command just exits Leo, returning to the command line (if run from a console), and Leo does not restart.

**Comment by edreamleo** [comment]
Fixed via PR #3879.
